adal/authority.py

```diff
@@ -70,8 +70,11 @@ def _validate_authority_url(self):
         if self._url.query:
             raise ValueError("The authority url must not have a query string.")
 
-        if self._url.path.count('/') > 1:
+        path_parts = [part for part in self._url.path.split('/') if part]
+        if len(path_parts) > 1:
             raise ValueError("The authority url must be of the format https://login.microsoftonline.com/your_tenant")
+        elif len(path_parts) == 1:

```

Comment:
There can be a case when the length is 0, in which case we would want to handle it differently and not normalize it. So I think the `elif ...` should be there.

Reply:
We discussed offline and clarified understandings. Since the `rstrip(...)` can handle case when length is 0 or more, the code could be written this conceptually different way:

```python
        if len(path_parts) > 1:
            raise ValueError("The authority url must be of the format https://login.microsoftonline.com/your_tenant")
        self._url = urlparse(self._url.geturl().rstrip('/'))
```

yet the actual behavior would be the same.